### PR TITLE
Add Auth Context Provider

### PR DESCRIPTION
**Pull request by tjosepo**
# Description

I finally understood how to use React's esoteric Context Providers and I wanted to see if I could create one to deal with the authentication on the frontend.

This PR removes the authentication logic that was split up in different files and groups it all into the ``AuthProvider``.

Children elements can use the ``useAuth`` hook to log in, log out, check if the user is logged in, etc.

## Type of change

- [X] CHORE (Something that needs to be done outside of the issue)

# Checklist:

- [X] My pull request references the issue in the comments
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/contexts/Auth.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+type Response = ErrorResponse | SuccessResponse;
+
+export const AuthProvider = ({ children }: Props) => {
+  const [isLoggedIn, setLoggedIn] = useState<boolean>();
+
+  const logIn = async (email: string, password: string) => {
+    const basicAuthToken = getBasicAuth(email, password);
+
+    const request = await fetch('http://localhost:5000/user/login/', {
```
Comment (phong1233) [inline frontend/src/contexts/Auth.tsx]: Would be nice to have the hostname in the configs or env

File: frontend/src/pages/login/Login.tsx [hunk 2]
```diff
@@ -1,25 +1,13 @@
...
-interface SuccessResponse {
-  status: true;
-  name: string;
-  token: string;
-}
+import { useAuth } from '../../contexts/Auth';
+import './Login.scss';
 
-export default function LoginV2({ setLoggedIn }: Props) {
+export default function LoginV2() {
```
Comment (phong1233) [inline frontend/src/pages/login/Login.tsx]: Not sure if the v2 was intended?

**Review by phong1233 — APPROVED** [review]
Exuberant! left minor comments otherwise looks good to me

File: frontend/src/contexts/Auth.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    return response;
+  };
+
+  const getBasicAuth = (email: string, password: string): string => {
+    const hash = btoa(`${email}:${password}`);
+    const token = `Basic ${hash}`;
+    return token;
+  };
```
Comment (sebastien-blain) [inline frontend/src/contexts/Auth.tsx:55]: Variable names could use some work. I am aware that your input is a token and your output is another token, so names are going to get ugly either way.

**Review by sebastien-blain — APPROVED** [review]
Nice job Tommy, it does look better this way.

File: frontend/src/contexts/Auth.tsx (hunk 3 above)
Comment (tjosepo) [reply frontend/src/contexts/Auth.tsx:55]: Don't try to pull a UNO Reverse card on me. In this context, the variable names are accurate.

File: frontend/src/pages/login/Login.tsx (hunk 2 above)
Comment (tjosepo) [reply frontend/src/pages/login/Login.tsx]: Woopsies. I meant to remove that and forgot.

File: frontend/src/contexts/Auth.tsx (hunk 1 above)
Comment (tjosepo) [reply frontend/src/contexts/Auth.tsx]: I'll look into it.

File: frontend/src/Api.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const api = process.env.REACT_APP_API;
+
+export const API_USER_LOGIN = `${api}/user/login`;
+export const API_USER_FORGOT = `${api}/user/forgot`;
+export const API_USER_RESET = `${api}/user/reset`;
```
Comment (tjosepo) [inline frontend/src/Api.tsx]: @phong1233 Tell me what you think of this approach.

Unfortunately, I wasn't able to use the .env file from the env/ folder.  facebook/create-react-app#4283 seems to indicate that you can't select where the .env file is located unless you eject from React.

Comment (phong1233) [reply frontend/src/Api.tsx]: This is how I did it in the backend https://github.com/Talos10/SOEN390-team-9/blob/main/backend/config.ts
The env file is only used when launching using docker

Comment (tjosepo) [reply frontend/src/Api.tsx]: Trueeee.
I'll adjust it.

**Pull request merged**